clients/python/tests/test_client.py

```diff
@@ -1206,3 +1208,129 @@ async def test_as_mlops_engineer_i_would_like_to_store_a_longer_documentation_fo
     )
     assert da
     assert da.uri == "https://README.md"
+
+
+class MockPath:
+    def __init__(self, path_arg, file_contents=None):
+        self.path_arg = str(path_arg)
+        self.file_contents = file_contents or {}
+
+    def exists(self):
+        print("exists", self.path_arg)
+        return self.path_arg in self.file_contents
+
+    def read_text(self):
+        return self.file_contents.get(self.path_arg, "<wrong, returning arbitrary string>")
+
+    def __str__(self):
+        return self.path_arg
+
+    def __repr__(self):
+        return f"MockPath('{self.path_arg}', file_contents={self.file_contents})"
+
+
+class TestUserToken:
+    @patch.object(ModelRegistry, "get_registered_models")
+    def test_user_token_from_envvar(self, mock_get_registered_models):
+        """Test for user not providing explicitly user_token,
+        reading user token from environment variable."""
+        test_token = "test-token-from-envvar" # noqa: S105
+
+        with tempfile.NamedTemporaryFile(mode="w", delete=False) as token_file:
+            token_file.write(test_token)
+            token_file_path = token_file.name
+
+        try:
+            with patch.dict(os.environ, {"KF_PIPELINES_SA_TOKEN_PATH": token_file_path}):
+                client = ModelRegistry(
+                    server_address="http://localhost",
+                    port=8080,
+                    author="test_author",
+                    is_secure=False,
+                    # user_token=None -> ... Let it read from Env var
+                )
+                assert client is not None
+                assert client._api.config.access_token == test_token
+        finally:
+            os.unlink(token_file_path)
+
+    @patch.object(ModelRegistry, "get_registered_models")
+    def test_user_token_from_k8s_file(self, mock_get_registered_models):
+        """Test for user not providing explicitly user_token,
+        reading user token from Kubernetes service account token file."""
+        test_token = "test-token-from-k8s-file" # noqa: S105
+        k8s_token_path = "/var/run/secrets/kubernetes.io/serviceaccount/token" # noqa: S105
+
+        with patch("model_registry._client.Path") as mock_path_class, \
```

Comment:
This seems a little fragile. Maybe it's okay given it's **only** for a call to the `ModelRegistry` constructor (and you've mocked only for the `model_registry._client` module), but it could cause a small headache later if we add anything else that uses `Path` (or even some builtin utility that uses `Path` internally) in the constructor. I also hesitate to set the precedent.

Instead of mocking `Path` could we maybe set a top-level variable for the path value? (i.e. at the same place we're setting `DEFAULT_USER_TOKEN_ENVVAR` we could set `USER_TOKEN_PATH = "/var/run/secrets/kubernetes.io/serviceaccount/token"`). Then you could simply patch this variable to be the path of a temp file that you write to.

Reply:
done with https://github.com/kubeflow/model-registry/pull/1553/commits/60db060d4c78ebf0c89a36157d2262453d2f30df
and look overall simpler, do you agree?